File: src/main/java/hilfling/backend/hilfling/controllers/CategoryController.java
package hilfling.backend.hilfling.controllers;

import hilfling.backend.hilfling.model.Category;
import hilfling.backend.hilfling.service.CategoryService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/v1/categories")
public class CategoryController extends GenericBaseControllerImplementation<Category>{
    @Autowired
    private CategoryService service;
    public CategoryService getService() {
        return service;
    }
}

File: src/main/kotlin/hilfling/backend/hilfling/controller/CategoryController.kt
package hilfling.backend.hilfling.controller

import hilfling.backend.hilfling.model.*
import hilfling.backend.hilfling.repository.CategoryRepository
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/categories")
class CategoryController {
    val repository = CategoryRepository()

    @GetMapping("/{id}")
    fun getById(@PathVariable("id") id: Int): Category? {
        return repository.findById(id)
    }

    @GetMapping
    fun getAll(): List<Category> {
        return repository.findAll()
    }

    @PostMapping
    fun create(
            @RequestParam("name") name: String
    ): Category {
        return repository.create(
                name
        )
    }
}